R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -645,16 +645,17 @@ test_that("test tableNames and tables", {
   df <- read.json(jsonPath)
   createOrReplaceTempView(df, "table1")
   expect_equal(length(tableNames()), 1)
-  tables <- tables()
+  tables <- listTables()
```

Comment:
right, there are some differences of the output (most notability catalog.listTables returns a Dataset<Table> - but I'm converting that into a DataFrame anyway), and I thought list* would be more consistent with other methods like listColumn, listDatabases()

```
> head(tables("default"))
  database tableName isTemporary
1  default      json       FALSE
Warning message:
'tables' is deprecated.
Use 'listTables' instead.
See help("Deprecated")
> head(listTables("default"))
  name database description tableType isTemporary
1 json  default        <NA>  EXTERNAL       FALSE
```

If you think it makes sense, we could make `tables` an alias of `listTables` - it's going to call slightly different code on the Scala side and there are new columns and one different column name being returned.